packages/react-reconciler/src/ReactFiberHooks.js

```diff
@@ -508,14 +504,31 @@ export function bailoutHooks(
   }
 }
 
-export function resetHooks(): void {
+export function resetHooksAfterThrow(): void {
   // We can assume the previous dispatcher is always this one, since we set it
   // at the beginning of the render phase and there's no re-entrancy.
   ReactCurrentDispatcher.current = ContextOnlyDispatcher;
 
-  // This is used to reset the state of this module when a component throws.
-  // It's also called inside mountIndeterminateComponent if we determine the
-  // component is a module-style component.
+  if (didScheduleRenderPhaseUpdate) {
+    const current = (currentlyRenderingFiber: any).alternate;
```

Comment:
@acdlite What if this is render phase updates during initial mount?

Reply:
It doesn't matter then because either the whole fiber will get thrown out. Except in legacy mode goddamn it. I'll add a test in the other PR.